1Password/pulumi-onepassword
# Build provider locally by configuring .goreleaser.yml file

This PR implements the work described in the [Building the Provider Locally](https://github.com/pulumi/pulumi-tf-provider-boilerplate#building-the-provider-locally) section of the repo's README.md file.

## Notes

- The "Building the Provider Locally" section linked above instructs us to run the `goreleaser build --rm-dist --skip-validate` command, but both the `--rm-dist` and `--skip-validate` flags are deprecated; see testing steps below for the new option names.
- The testing steps below also include the `--snapshot` flag; not passing this option results in `error=git doesn't contain any tags`, which makes sense since the provider has not yet been released.

## How to Test

1. Clone the `pulumi-onepassword` repo locally and run `git pull && git switch ruetz-build-provider-locally` to view/test this branch's changes.
2. Run the `make provider` command.
   - [x] Verify that the command succeeds and does not exit with an error.
3. Run the `which goreleaser` command; if you receive a "goreleaser not found" error, run `brew install goreleaser` to install it.
4. Run the `goreleaser build --clean --skip=validate --snapshot` command.
   - [x] Verify that the `dist/` directory is created (note that it isn't committed to source control because it's listed in the `.gitignore` file).
   - [x] Verify that the `dist/` directory is populated with the matrix of binaries; there should be six files total, i.e. both amd64 and arm64 architectures for Darwin (macOS), Linux, and Windows.
